analyzers/tests/SonarAnalyzer.Test/Rules/PasswordsShouldBeStoredCorrectlyTest.cs

```diff
@@ -0,0 +1,149 @@
+﻿/*
+ * SonarAnalyzer for .NET
+ * Copyright (C) 2015-2024 SonarSource SA
+ * mailto: contact AT sonarsource DOT com
+ *
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public
+ * License as published by the Free Software Foundation; either
+ * version 3 of the License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with this program; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ */
+
+using SonarAnalyzer.Rules.CSharp;
+
+namespace SonarAnalyzer.Test.Rules;
+
+[TestClass]
+public class PasswordsShouldBeStoredCorrectlyTest
+{
+    private static readonly VerifierBuilder Builder = new VerifierBuilder<PasswordsShouldBeStoredCorrectly>();
+
+#if NET
+    [TestMethod]
+    public void PasswordsShouldBeStoredCorrectly_CS_Core() =>
+        Builder
+            .WithOptions(ParseOptionsHelper.FromCSharp8)
+            .AddPaths("PasswordsShouldBeStoredCorrectly.Core.cs")
+            .AddReferences([
+                AspNetCoreMetadataReference.MicrosoftExtensionsIdentityCore,
+                AspNetCoreMetadataReference.MicrosoftAspNetCoreCryptographyKeyDerivation,
+                CoreMetadataReference.SystemSecurityCryptography,
+            ])
+            .Verify();
+#endif
+
+    [TestMethod]
+    public void PasswordsShouldBeStoredCorrectly_CS_Rfc2898DeriveBytes() =>
```

Comment:
This test covers ONLY the constructor of `Rfc2898DeriveBytes`.
The case of `Rfc2898DeriveBytes.Pkbdf2` invocation exists only in the core testcase, since the method does not exist in .NET Framework.